assets/js/atomic/utils/register-block-single-product-template.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { getBlockType } from '@wordpress/blocks';
+import { subscribe, select } from '@wordpress/data';
+
+export const registerBlockSingleProductTemplate = ( {
+	registerBlockFn,
+	unregisterBlockFn,
+	blockName,
+}: {
+	registerBlockFn: () => void;
+	unregisterBlockFn: () => void;
+	blockName: string;
+} ) => {
+	let currentTemplateId: string | undefined;
+
+	subscribe( () => {
```

Comment:
Maybe we can add 2nd argument to subscribe function i.e. storeNameOrDescriptor parameter, the listener function will be called only on updates on that one specific registered store. In our case, I believe it would be `core/edit-site`. 
What do you think?

Reply:
Maybe I don't understand what do you mean, but the `subscribe` function receive only the callback as parameter.

Reply:
Sorry for not being clear.

I checked [this doc](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#subscribe) and it looks like there is 2nd argument that could be passed. Here is the text from the link:

```
Given a listener function, the function will be called any time the state value
of one of the registered stores has changed. If you specify the optional
storeNameOrDescriptor parameter, the listener function will be called only
on updates on that one specific registered store.

This function returns an unsubscribe function used to stop the subscription.

Usage

import { subscribe } from '@wordpress/data';

const unsubscribe = subscribe( () => {
    // You could use this opportunity to test whether the derived result of a
    // selector has subsequently changed as the result of a state update.
} );

// Later, if necessary...
unsubscribe();

Parameters

    listener Function: Callback function.
    storeNameOrDescriptor string|StoreDescriptor?: Optional store name.
```

Please let me know if I am missing something.

Reply:
Thanks for creating a PR on DefinitelyTyped repo 🙌🏻